IdentityCore/src/pop_manager/MSIDDevicePopManager.m

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDDevicePopManager.h"
+#import "MSIDConstants.h"
+#import "NSData+MSIDExtensions.h"
+#import "MSIDJWTHelper.h"
+#import "MSIDAssymetricKeyGeneratorFactory.h"
+#import "MSIDAssymetricKeyLookupAttributes.h"
+#import "MSIDAssymetricKeyPair.h"
+#import "MSIDDefaultTokenCacheAccessor.h"
+
+static NSString *jwkTemplate = @"{\"e\":\"%@\",\"kty\":\"RSA\",\"n\":\"%@\"}";
+static NSString *kidTemplate = @"{\"kid\":\"%@\"}";
+
+@interface MSIDDevicePopManager()
+
+@property (nonatomic) MSIDCacheConfig *cacheConfig;
+@property (nonatomic) MSIDDefaultTokenCacheAccessor *tokenCache;
+@property (nonatomic) id<MSIDAssymetricKeyGenerating> keyGeneratorFactory;
+@property (nonatomic) MSIDAssymetricKeyLookupAttributes *keyPairAttributes;
+@property (nonatomic) MSIDAssymetricKeyPair *keyPair;
+@property (nonatomic) NSString *requestConfirmation;
+@property (nonatomic) NSString *kid;
+
+@end
+
+@implementation MSIDDevicePopManager
+
+- (instancetype)initWithCacheConfig:(MSIDCacheConfig *)cacheConfig cache:(MSIDDefaultTokenCacheAccessor *)tokenCache
+{
+    self = [super init];
+    if (self)
+    {
+        _cacheConfig = cacheConfig;
+        _tokenCache = tokenCache;
+    }
+    
+    return self;
+}
+
+ - (id<MSIDAssymetricKeyGenerating>)keyGeneratorFactory
```

Comment:
are there any thread safety concerns?

Reply:
I believe we are fine because we read the same key pair from the keychain everytime and all these values are read from the same key pair and not set to different values any time.

Reply:
As discussed. Resolving this.